mlptrain/sampling/tests/test_bias.py

```diff
@@ -74,4 +84,34 @@ def test_bias():
     assert os.path.exists('tmp.xyz')
 
 
+@work_in_tmp_dir()
+def test_differencedistance():
+    """Test the DifferenceDistance class for reaction coordinate"""
+
+    system = mlt.System(_h2o(), box=[50, 50, 50])
+
+    config = system.random_configuration()
+    atoms = config.ase_atoms
+
+    diff_dist = mlt.DifferenceDistance((0, 1), (0, 2))
+
+    # Reaction coordinate should contain two pairs of atoms
+    assert len(diff_dist.atom_pair_list) == 2
+
+    rxn_coord = diff_dist(atoms)
+
+    # Check calling the class returns the reaction coordinate
+    assert np.isclose(rxn_coord, 0.614, 0.1)
+
+    grad = diff_dist.grad(atoms)
+
```

Comment:
Would be nice to assert this is the same as the numerical gradient

Reply:
I've asserted it is close to the correct answer, but not sure how I could check it to the numerical gradient since the function is a combination of the 3 atoms

Reply:
will attempt to implement what I mean 😅